src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7374,16 +7371,18 @@ class C
 {
     static void Main()
     {
+        Test("""");
         Test(""test string"");
         Test(""test string? I think not!"");
         Test(""WrongString"");
     }
     static void Test(ReadOnlySpan<char> chars) => Console.WriteLine(chars is ""test string"");
 }
 ";
-            var compilation = CreateCompilationWithSpanAndMemoryExtensions(source, options: TestOptions.DebugExe, parseOptions: TestOptions.RegularPreview)
```

Comment:
What's the reason we're not testing on desktop TFM anymore (using our own implementation of `MemoryExtensions`)?

Reply:
The reason for testing with `TargetFramework.NetCoreApp` rather than our own implementation of `Span<T>` and `MemoryExtensions` is to ensure we're testing with type declarations that match the actual .NET API.

The .NET implementation of `Span<T>` includes a conversion operator from `Span<T>` to `ReadOnlySpan<T>` for instance, and the .NET implementation of `ReadOnlySpan<T>` does not include a conversion operator from `string` to `ReadOnlySpan<T>`.

Reply:
I understand wanting the majority of our tests to be .NET Core only, after all that is where the official support is. At the same time though this feature is going to be used on .NET Framework. Consider concretely that the compiler is going to use this as it represents a measurable perf gain for VS IDE scenarios (removes allocations during command line parsing). Given that I think we need at least a few smoke tests on .NET Framework to make sure we have a bit of sanity coverage.

Reply:
I've changed most tests to run on both .NET Framework and .NET Core. With .NET Core, the tests use the runtime assemblies rather than TestSources.

Reply:
Sounds good. Thanks!